toolchain/check/subst.h

```diff
@@ -0,0 +1,35 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_CHECK_SUBST_H_
+#define CARBON_TOOLCHAIN_CHECK_SUBST_H_
+
+#include "toolchain/check/context.h"
+#include "toolchain/sem_ir/ids.h"
+
+namespace Carbon::Check {
+
+// A substitution that is being performed.
+struct Substitution {
+  // The ID of a `BindSymbolicName` instruction that is being replaced.
+  SemIR::InstId bind_id;
+  // The replacement constant value to substitute.
+  SemIR::ConstantId replacement_id;
+};
+
+using Substitutions = llvm::ArrayRef<Substitution>;
+
+// Replaces the `BindSymbolicName` instruction `bind_id` with `replacement_id`
+// throughout the constant `const_id`, and returns the substituted value.
+auto SubstConstant(Context& context, SemIR::ConstantId const_id,
```

Comment:
It feels odd to me that you do `Substitution` but `Subst`. `Subst` seems like an unusual abbreviation (I'm used to things like `re.sub` from Python); maybe just write out `Substitute`?

On locals I don't think about it as much as on function names, FWIW. Maybe I'll just get used to it. But then it's probably an abbreviation to add to the arch doc.

Reply:
I already added it to the arch doc. Maybe it seems unsurprising to me because it's used heavily in Clang. How about we check back and see if you've got used to it; if it remains weird then we can unabbreviate it. Based on experience with explorer, I think this is going to be something we do a lot, though maybe that also argues for making the name `Substitutions` shorter.